src/Turtle/Bytes/Lazy.hs

```diff
@@ -0,0 +1,639 @@
+    let p' = p
+            { Process.std_in  = Process.CreatePipe
+            , Process.std_out = Process.CreatePipe
+            , Process.std_err = Process.CreatePipe
+            }
+
+    let open = do
+            (Just hIn, Just hOut, Just hErr, ph) <- liftIO (Process.createProcess p')
+            System.IO.hSetBuffering hIn (System.IO.BlockBuffering Nothing)
+            return (hIn, hOut, hErr, ph)
+
+    -- Prevent double close
+    mvar <- MVar.newMVar False
+    let close handle = do
+            MVar.modifyMVar_ mvar (\finalized -> do
+                Control.Monad.unless finalized
+                    (ignoreSIGPIPE (System.IO.hClose handle))
+                return True )
+
+    Exception.bracket open (\(hIn, _, _, ph) -> close hIn >> Process.terminateProcess ph) (\(hIn, hOut, hErr, ph) -> do
+        let feedIn :: (forall a. IO a -> IO a) -> IO ()
+            feedIn restore =
+                restore (ignoreSIGPIPE (outhandle hIn s))
+                `Exception.finally` close hIn
+
+        runConcurrently $ (,,)
+            <$> Concurrently (Exception.mask (\restore ->
+                    Async.withAsync (feedIn restore) (\a ->
+                        restore (Process.waitForProcess ph) <* halt a ) ))
+            <*> Concurrently (Data.ByteString.Lazy.hGetContents hOut)
+            <*> Concurrently (Data.ByteString.Lazy.hGetContents hErr) ) )
+
+{-| Run a command using @execvp@, streaming @stdout@ as chunks of `ByteString`
+
+    The command inherits @stderr@ for the current process
+-}
+inproc
+    :: Text
+    -- ^ Command
+    -> [Text]
+    -- ^ Arguments
+    -> Shell ByteString
+    -- ^ Chunks of bytes written to process input
+    -> Shell ByteString
+    -- ^ Chunks of bytes read from process output
+inproc cmd args =
+    stream (Process.proc (Data.Text.unpack cmd) (map Data.Text.unpack args))
+
+{-| Run a command line using the shell, streaming @stdout@ as chunks of
+    `ByteString`
+
+    This command is more powerful than `inproc`, but highly vulnerable to code
+    injection if you template the command line with untrusted input
+
+    The command inherits @stderr@ for the current process
+-}
+inshell
+    :: Text
+    -- ^ Command line
+    -> Shell ByteString
+    -- ^ Chunks of bytes written to process input
+    -> Shell ByteString
+    -- ^ Chunks of bytes read from process output
+inshell cmd = stream (Process.shell (Data.Text.unpack cmd))
+
+waitForProcessThrows :: Process.ProcessHandle -> IO ()
+waitForProcessThrows ph = do
+    exitCode <- Process.waitForProcess ph
+    case exitCode of
+        ExitSuccess   -> return ()
+        ExitFailure _ -> Exception.throwIO exitCode
+
+{-| `stream` generalizes `inproc` and `inshell` by allowing you to supply your
+    own custom `CreateProcess`.  This is for advanced users who feel comfortable
+    using the lower-level @process@ API
+
+    Throws an `ExitCode` exception if the command returns a non-zero exit code
+-}
+stream
```

Comment:
I don't think you need any of the streaming functions like `stream`/`streamWithErr` or the functions downstream of them.  The `systemStrict` and `systemStrictWithErr` versions already stream since they are using `Data.ByteString.Lazy.hGetContents`